processor/src/main/java/org/mapstruct/ap/internal/model/source/TargetReference.java

```diff
@@ -195,8 +195,7 @@ public TargetReference build() {
                 boolean isLast = i == entryNames.length - 1;
                 boolean isNotLast = i < entryNames.length - 1;
                 if ( isWriteAccessorNotValidWhenNotLast( targetWriteAccessor, isNotLast )
-                    || isWriteAccessorNotValidWhenLast( targetWriteAccessor, targetReadAccessor, mapping, isLast )
-                    || ( isNotLast && targetReadAccessor == null ) ) {
```

Comment:
hmm. I'm thinking why we (I? :) ) put this here.. Don't have a clear idea anymore..